src/azure-cli/azure/cli/command_modules/vm/_actions.py

```diff
@@ -40,15 +40,28 @@ def load_images_thru_services(cli_ctx, publisher, offer, sku, location):
         location = get_one_of_subscription_locations(cli_ctx)
 
     def _load_images_from_publisher(publisher):
-        offers = client.virtual_machine_images.list_offers(location, publisher)
+        from msrestazure.azure_exceptions import CloudError
+        try:
+            offers = client.virtual_machine_images.list_offers(location, publisher)
+        except CloudError as e:
+            logger.warning(str(e))
+            return
         if offer:
             offers = [o for o in offers if _matched(offer, o.name)]
         for o in offers:
-            skus = client.virtual_machine_images.list_skus(location, publisher, o.name)
+            try:
+                skus = client.virtual_machine_images.list_skus(location, publisher, o.name)
+            except CloudError as e:
+                logger.warning(str(e))
+                continue
             if sku:
                 skus = [s for s in skus if _matched(sku, s.name)]
             for s in skus:
-                images = client.virtual_machine_images.list(location, publisher, o.name, s.name)
+                try:
+                    images = client.virtual_machine_images.list(location, publisher, o.name, s.name)
+                except CloudError as e:
+                    logger.warning(str(e))
```

Comment:
e.message? I'm not familiar with it. What's the difference between `e.message` and `str(e)`? @jiasli

Reply:
As I know, str(object) will call the object.__str__() at last. and e.message is just a property of e. Whether these two values are the same depends on how to implement the function __str__(). If it return message directly, they will be the same. However, for most cases, __str__ will serialize all the properties of the object. @MyronFanQiu

Reply:
str(e) can tell more details including which publisher cannot be found. It's ok to use e.message here. The difference depends on specific implementation.
code:
msrestazure\azure_exceptions.py, class CloudError(ClientException):